rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -614,11 +570,31 @@ rmw_create_publisher(
       allocator->deallocate(type_hash_c_str, allocator->state);
     });
 
-  z_owned_keyexpr_t keyexpr = ros_topic_name_to_zenoh_key(
-    node->context->actual_domain_id,
-    topic_name,
-    publisher_data->type_support->get_name(),
-    type_hash_c_str);
+  const z_id_t zid = z_info_zid(z_loan(node->context->impl->session));
+
+  publisher_data->entity = rmw_zenoh_cpp::liveliness::Entity::make(
+    zid,
+    std::to_string(node_data->id),
+    std::to_string(
+      context_impl->get_next_entity_id()),
+    rmw_zenoh_cpp::liveliness::EntityType::Publisher,
+    rmw_zenoh_cpp::liveliness::NodeInfo{
+      node->context->actual_domain_id, node->namespace_, node->name, context_impl->enclave},
+    rmw_zenoh_cpp::liveliness::TopicInfo{
+      node->context->actual_domain_id,
+      rmw_publisher->topic_name,
+      publisher_data->type_support->get_name(),
+      type_hash_c_str,
+      publisher_data->adapted_qos_profile}
+  );
+  if (publisher_data->entity == nullptr) {
+    RMW_ZENOH_LOG_ERROR_NAMED(
+      "rmw_zenoh_cpp",
+      "Unable to generate keyexpr for liveliness token for the publisher.");
```

Comment:
add `rmw_publisher->topic_name` in the error message?

Reply:
Done https://github.com/ros2/rmw_zenoh/pull/269/commits/aef564483916cb723f56a22edb1d935ec89afce0